demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,694 @@
+/*
+ * FreeRTOS V202011.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/* Standard includes. */
+#include <stdlib.h>
+#include <string.h>
+
+/* Demo config. */
+#include "defender_demo_config.h"
+
+/* Metrics collector. */
+#include "metrics_collector.h"
+
+/* Report builder. */
+#include "report_builder.h"
+
+/* MQTT operations. */
+#include "mqtt_demo_helpers.h"
+
+/* JSON Library. */
+#include "core_json.h"
+
+/* Device Defender Client Library. */
+#include "defender.h"
+
+/**
+ * @brief Predefined thing name.
+ *
+ * This is the example predefine thing name and could be compiled in ROM code.
+ */
+#define THING_NAME                        clientcredentialIOT_THING_NAME
+
+/**
+ * @brief The length of #THING_NAME.
+ */
+#define THING_NAME_LENGTH                 ( ( uint16_t ) ( sizeof( THING_NAME ) - 1 ) )
+
+/**
+ * @brief Number of seconds to wait for the response from AWS IoT Device
+ * Defender service.
+ */
+#define DEFENDER_RESPONSE_WAIT_SECONDS    ( 2 )
+
+/**
+ * @brief The maximum number of times to run the loop in this demo.
+ */
+#ifndef DEFENDER_MAX_DEMO_COUNT
+    #define DEFENDER_MAX_DEMO_COUNT    ( 3 )
```

Comment:
I'd think a name like `DEFENDER_MAX_DEMO_TRY_COUNT` might be more descriptive, since it's the number of times to to try the demo rather than the number of times the demo is run.